module/data/item/spell.mjs

```diff
@@ -141,4 +134,19 @@ export default class SpellData extends SystemDataModel.mixin(
   get proficiencyMultiplier() {
     return 1;
   }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Provide a backwards compatible getter for accessing `components`.
+   * @deprecated since v2.5.
+   * @type {Set<string>}
+   */
+  get components() {
+    foundry.utils.logCompatibilityWarning(
+      "The `system.components` property has been deprecated in favor of a standardized \`system.properties\` property.",
+      { since: "DnD5e 2.5", until: "DnD5e 2.7", once: true }
+    );
+    return this.properties;
```

Comment:
I think the caller would expect back an object of booleans if they're still using the old `components` property.

```suggestion
    return this.properties.reduce((obj, p) => obj[p] = true, {});
```

Don't forget to adjust the JSDoc if you incorporate this suggestion.